### PR TITLE
fixes bug where querying by chatId results in no records

### PR DESCRIPTION
**Pull request by Jaredude**
previous logic for where clause of memoryType checked for existence of chatId and would query for memoryType is null This where clause logic results in an empty result for any request to the endpoint that filters by chatId

Getting messages without any filters
Screenshot 2024-02-12 at 8 14 32 PM

filtering by known chatId value using existing where clause
Screenshot 2024-02-12 at 8 16 09 PM

filtering by known chatId value using new where clause logic for memoryType
Screenshot 2024-02-12 at 8 15 09 PM

### CODE REVIEW

**Review by HenryHengZJ — APPROVED** [review]

File: packages/server/src/index.ts [hunk 1]
```diff
@@ -1453,7 +1453,7 @@ export class App {
                 chatflowid,
                 chatType,
                 chatId,
-                memoryType: memoryType ?? (chatId ? IsNull() : undefined),
+                memoryType: memoryType ?? undefined,
```
Comment (HenryHengZJ) [inline packages/server/src/index.ts:1456]: couldn't remember why we have the `isNull()` in the first place, but would you be able to check from the View Message dialog on UI, that it still works as expected

Comment (Jaredude) [reply packages/server/src/index.ts:1456]: It works as expected.
https://github.com/FlowiseAI/Flowise/assets/10228950/481bf246-e62a-4ec9-b91f-7ac73fd8c6b5

I found a bug with the way the chatType is filtered. I'll add another commit to this PR to resolve that issue as well.

**Comment by Jaredude** [comment]
@HenryHengZJ It turned out that the bug I thought was related to the source type was not an issue with source type at all but rather an issue with the date filter. If an invalid date was passed in for startDate or endDate, getChatMessage was not checking whether new Date() was resulting in an invalid date or not. The date picker control on the start date is defaulting to a epoch timestamp that's not compatible with new Date(), so the bug only happens when neither of the date fields are changed.

Bug (causing server crash):
https://github.com/FlowiseAI/Flowise/assets/10228950/46116b2c-1d9f-4cc3-9936-94b40a2375ea

Fixed:
https://github.com/FlowiseAI/Flowise/assets/10228950/a18b3856-9410-45a3-86bb-71c23a2f398c

date filtering working with >= <= logic instead of between to handle more robust date filtering:
https://github.com/FlowiseAI/Flowise/assets/10228950/a13cfe9c-5600-47b2-a126-f397b149fcd4

**Comment by HenryHengZJ** [comment]
@Jaredude I'm confused how to replicate the issue, it seems to be working for me when filter by dates:
image

**Comment by Jaredude** [comment]
@HenryHengZJ I think the key to reproduce might be in live reload vs hot reload and the initial state of the date fields. Here's my steps to reproduce:
1. yarn build
2. yarn start
3. go into flow
4. view messages
5. Filter by type

The above steps will crash the server.

I am able to reliably reproduce using the steps above so long as I don't change the start date first. Let me know if you're unable to reproduce using the steps above, and I'll look at things again.
https://github.com/FlowiseAI/Flowise/assets/10228950/1b9f7a5b-282d-44a5-980f-4976fdc1982c

**Comment by HenryHengZJ** [comment]
> @HenryHengZJ Here's my steps to reproduce:
> 
> 1. yarn build
> 2. yarn start
> 3. go into flow
> 4. view messages
> 5. Filter by type
> 
> The above steps will crash the server.
> 
> Changing the start date will not result in a crash, as it will provide a date in the proper format at that point. I am able to reliably reproduce using the steps above so long as I don't change the start date first. Let me know if you're unable to reproduce using the steps above, and I'll look at things again. [video example of server crash](https://github.com/FlowiseAI/Flowise/assets/10228950/1b9f7a5b-282d-44a5-980f-4976fdc1982c)
> 
> Edit: You can also cause a crash by passing an invalid date to the start date via the API.

Which branch are you on? I've tried multiple times on the `main` branch and never bumped into any issues. Filter by source works just fine

**Comment by Jaredude** [comment]
I'm on main. I just did a pull, to make sure I'm up to date, and there were no files updated.

I thought the issue was filtering by source, but the issue is really with the handling of the date fields. Currently, we don't check whether new Date(startDate) or new Date(endDate) produce valid dates, so when an invalid date value gets passed into the .find method, the server will crash.

In my browser (Chrome or Safari), when I don't change the value for start date the call to new Date(startDate) is converting to "Invalid Date". This looks like it's a localization issue (possibly with the date picker library). But the server should be performing the date validation anyway since it implicitly knows the field must be a valid date to include it in the sql query.

Without changing the value of the start date the server will crash even by changing the end date and note even the type. [Video of server crash on end date change without start date change](https://github.com/FlowiseAI/Flowise/assets/10228950/4db8ab72-0b82-4826-8d7f-7bbab72e8c9f).

This crash can also be produced on Postman using any invalid values for either startDate or endDate.
Screenshot 2024-02-15 at 12 11 57 PM

**Review by HenryHengZJ — APPROVED** [review]

**Comment by Jaredude** [comment]
Whew! 😅I was starting to worry something was wrong with my laptop!

**Pull request merged**